provider/cloudflare/cloudflare.go

```diff
@@ -455,14 +449,19 @@ func (p *CloudFlareProvider) submitChanges(ctx context.Context, changes []*cloud
 					failedChange = true
 					log.WithFields(logFields).Errorf("failed to delete record: %v", err)
 				}
-				chID, oldCh := p.getCustomHostnameIDbyOrigin(chs, change.ResourceRecord.Name)
+				if change.CustomHostname.Hostname == "" {
+					continue
+				}
+				log.WithFields(logFields).Infof("Deleting custom hostname %v", change.CustomHostname.Hostname)
+				chID, _ := p.getCustomHostnameOrigin(chs, change.CustomHostname.Hostname)
 				if chID == "" {
+					log.WithFields(logFields).Infof("Custom hostname %v not found", change.CustomHostname.Hostname)
 					continue
 				}
 				chErr := p.Client.DeleteCustomHostname(ctx, zoneID, chID)
 				if chErr != nil {
 					failedChange = true
-					log.WithFields(logFields).Errorf("failed to delete custom hostname %v/%v: %v", chID, oldCh, chErr)
+					log.WithFields(logFields).Errorf("failed to delete custom hostname %v/%v: %v", chID, change.CustomHostname.Hostname, chErr)
 				}
```

Comment:
lines not covered with unit-tests
![Screenshot 2025-03-06 at 07 49 31](https://github.com/user-attachments/assets/48be5931-b5ff-4b37-9d64-a30226bd816c)